### PR TITLE
#3109 - Autotests: Reaction Auto-Mapping Tool

### PR DESCRIPTION
**Pull request by AKZhuk**
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [ ] unit-tests written
- [x] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

Fallen tests in pipeLine passed locally:
image

### CODE REVIEW

**Review by chgayane — APPROVED** [review]

File: ketcher-autotests/tests/utils/clicks/index.ts [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+  DELAY_IN_SECONDS,
+  ReactionMappingTool,
+  delay,
+  resetCurrentTool,
+  selectButtonById,
+  selectNestedTool,
+  takeEditorScreenshot,
+} from '..';
 import { AtomLabelType } from './types';
+// import { AutoMapOptions } from 'ketcher-core';
```
Comment (Nitvex) [inline ketcher-autotests/tests/utils/clicks/index.ts:15]: Can we remove this comment?

File: ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts [hunk 2]
```diff
@@ -1,27 +1,187 @@
...
+    /**
+     * Test case: EPMLSOPKET-1808
+     * Description:  UI dialog
+     */
+    await openFileAndAddToCanvas('reaction_2.rxn', page);
+    await selectNestedTool(page, ReactionMappingTool.AUTOMAP);
+    await takeEditorScreenshot(page);
+    await pressButton(page, 'Discard');
+  });
+  test.describe('full reaction on canvas', () => {
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts:71]: done

File: ketcher-autotests/tests/utils/clicks/index.ts [hunk 3]
```diff
@@ -137,3 +147,19 @@ export async function moveOnBond(
...
+  withScreenshot = true,
+) {
+  await resetCurrentTool(page);
+  await selectNestedTool(page, ReactionMappingTool.AUTOMAP);
+  await pressButton(page, 'Discard');
+  await selectOption(page, mode);
+  await selectButtonById('OK', page);
+  if (withScreenshot) {
+    await delay(DELAY_IN_SECONDS.ONE);
+    await takeEditorScreenshot(page);
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply ketcher-autotests/tests/utils/clicks/index.ts:159]: done

File: ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts [hunk 4]
```diff
@@ -1,27 +1,187 @@
...
+     * Test case: EPMLSOPKET-1801
+     * Description: Not possible when the reaction is absent on canvas
+     */
+    await selectNestedTool(page, ReactionMappingTool.AUTOMAP);
+    await takeEditorScreenshot(page);
+    await openFileAndAddToCanvas('four-structures.mol', page);
+    await selectNestedTool(page, ReactionMappingTool.AUTOMAP);
+    await takeEditorScreenshot(page);
+    await selectLeftPanelButton(LeftPanelButton.ReactionPlusTool, page);
+    await clickOnTheCanvas(page, -230, 0);
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts]: done

File: ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts [hunk 5]
```diff
@@ -1,27 +1,187 @@
...
+    await clickOnTheCanvas(page, 200, 0);
+    await selectNestedTool(page, ReactionMappingTool.AUTOMAP);
+  });
+
+  test('UI dialog', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1808
+     * Description:  UI dialog
+     */
+    await openFileAndAddToCanvas('reaction_2.rxn', page);
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts]: done

File: ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts [hunk 6]
```diff
@@ -1,27 +1,187 @@
...
+    const modes = ['Discard', 'Keep', 'Alter'];
+
+    for (const mode of modes) {
+      test(`${mode} mode`, async ({ page }) => {
+        await openFileAndAddToCanvas('reaction_3.rxn', page);
+        await applyAutoMapMode(page, mode);
+        await selectTopPanelButton(TopPanelButton.Undo, page);
+        await takeEditorScreenshot(page);
+        await selectLeftPanelButton(LeftPanelButton.ReactionMappingTool, page);
+        await clickOnTheCanvas(page, -250, 8);
```
[the remark this answers is not in the document]
Comment (AKZhuk) [reply ketcher-autotests/tests/Indigo-Tools/Reaction Auto-mapping tool/reaction-am-tool-veryfing-button.spec.ts]: done

File: ketcher-autotests/tests/utils/clicks/index.ts (hunk 1 above)
Comment (AKZhuk) [reply ketcher-autotests/tests/utils/clicks/index.ts]: done

**Review by Zhirnoff — APPROVED** [review]

**Pull request merged**